### ISSUE TITLE
Have a way to configure beep on error

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#12672]
First of all, sorry in case this issue is a duplicate of another one.

### Is your feature request related to a problem? Please describe.
In alpha and beta versions of NVDA, a beep is heard when an error is logged. I am missing this beep when I am doing some tests on a stable NVDA (add-on tests or tests of NVDA itself).

### Describe the solution you'd like

Add an option to configure when the error should make a beep with the 3 following choices:
* "Always": i.e. in NVDA stable as well as alpha/beta
* "Never": neither in NVDA stable nor in alpha/beta
* "Only in dev releases": beeps in alpha/beta but not in stable release. It is the current behaviour and should be the default.

The option may be located in the advanced settings panel.

### Describe alternatives you've considered

For now I have a global plugin in my scratchpad to do a similar job. But it's an issue when I make some tests with add-ons disabled.

### Additional context
N/A

**Comment by feerrenrut** [comment]
Happy for this to be added, but given alpha/beta are really intended for testing, and we want users to be aware of and then report errors, I don't think the "Never" option should be included.

**Comment by CyrilleB79** [comment]
Yes this makes sense. Let's remove the "Never" option.

**Comment by leonardder** [comment]
I really dig this idea, as this generally holds me back from stable releases, as I am that accustomed to error sounds that I really miss them in case there's a real error.

**Comment by leonardder** [comment]
Related to #11634

**Comment by lukaszgo1** [comment]
While I agree with this proposal in general I'd like to make a case for 'never' being an option. Consider the following real life example where having a way to suppress a beep on error can be useful for developers. After PR #11782 got merged accessing Internet explorer controls resulted in many errors from garbage Handler. Since the mailer that I am using displays messages in MSHTML controls I've been getting an error sounnd each time when the message was opened or closed so hardly an comfortable experience. In such case where the given issue has been reported and user awaits a fix it would be pretty nice to be able to disable error sounds in a configuration profile for a given program as the only alternative is to downgrade to the stable build.

**Comment by feerrenrut** [comment]
We really want to discourage relying on alpha versions of NVDA in "production" environments. If there are annoying errors like this, users should fall pack to a prior final release. Either have the prior stable release available as a portable install, or use the alpha as a portable install.

Disabling the error sound makes it too easy to forget to enable it again. We also don't want to train users to be complacent about error sounds, turning off error sounds means that errors in different situations won't be reported.
